manifests/v2/base/runtimes/pre-training/torch-distributed.yaml

```diff
@@ -0,0 +1,33 @@
+apiVersion: kubeflow.org/v2alpha1
+kind: ClusterTrainingRuntime
+metadata:
+  name: torch-distributed
+  labels:
+    training.kubeflow.org/phase: pre-training
+spec:
+  mlPolicy:
+    numNodes: 1
+    torch:
+      numProcPerNode: auto
+  template:
+    spec:
+      replicatedJobs:
+        - name: trainer-node
+          template:
+            spec:
+              template:
+                spec:
+                  containers:
+                    - name: trainer
+                      image: pytorch/pytorch:2.5.0-cuda12.4-cudnn9-runtime
```

Comment:
I was thinking how will the user configure PyTorch version of their choice?

Reply:
Data Scientists can always override the base image or Platform Engineers can configure runtime with the appropriate torch version.
From my past experience and discussions, usually ML teams try to stitch with a single PyTorch version for all of their ML Development.

Do you see other use-cases ?

Reply:
> From my past experience and discussions, usually ML teams try to stitch with a single PyTorch version for all of their ML Development.

In the production team, I think that you are mostly right. But, in the research department, each researcher often uses the different PyTorch versions, IMO.

But the discussion is a separate one. So I think that we can merge this PR in any case.